backend/services/apple_auth_service.py

```
@@ -0,0 +1,154 @@
+            response = requests.get(self.APPLE_KEYS_URL, timeout=10)
+            response.raise_for_status()
+            
+            self._apple_keys = response.json()
+            self._keys_cache_time = current_time
+            
+            logger.info("Successfully fetched Apple public keys")
+            return self._apple_keys
+            
+        except requests.ConnectionError as e:
+            logger.error(f"Connection failed while fetching Apple public keys: {str(e)}")
+            raise ConnectionError("Unable to connect to Apple servers")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)